### PR TITLE
Updated developer guide find

### PR DESCRIPTION
**Pull request by DarrenCsAcc**
Updated DG on find feature

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -320,9 +320,26 @@ The find operation is facilitated by `FindCommand`. It extends `Command` and imp
...
+4. Since a valid prefix for name is present `FindCommandParser#parse()` will then call `parsePredicates` with the argument `john`.
+
+5. In `parsePredicates` an empty is first created `applicantPredicateList` before, it checks for the type of prefix accompanying the argument which be of type  name, scholarship or major.
+
+6. After identifying the prefix is of type name it will call `getKeywordsList` to get the keyword for searching and in this case it will be `john`.
+
+7. After identifying the keyword `john` the method, `NameContainsKeywordsPredicate` will be invoked with the argument `john`.
+
+8. `NameContainsKeywordsPredicate` will check if any names in the main list has `john` present in their names.If `john` is present, that name in the main
+    list will be added to `applicantPredicateList`.All valid names are added to `applicantPredicateList`
```
Comment (L1uY1jun) [inline docs/DeveloperGuide.md]: Try to abstract it out more, dont speak about specific prefixes or keywords too much, check the edit applicant feature below for an example

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -366,6 +383,19 @@ The following activity diagram summarizes what happens when a user executes a ed
...
+For future iterations we plan to implement 2 new features.
+
+1. Sending of email to the applicants.
+This will help applicant be notified of their result more quickly.
+In this current iteration we still have to message the client manually.
+
+2. Exporting of Json file into Excel.
+The university admin staff might want to process or analyse the result in a more sophisticated manner.
+Excel is needed to aid in more sophisticated analysis as it has more functions.
+
```
Comment (L1uY1jun) [inline docs/DeveloperGuide.md]: rather than Future iterations write it like 
**Proposed features** - Coming soon in future iterations
Or smth like that

Comment (DarrenCsAcc) [reply docs/DeveloperGuide.md]: Ok done

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (DarrenCsAcc) [reply docs/DeveloperGuide.md]: Tried to work on it please have a look

**Comment by benjytan45678** [comment]
LGTM, merged

**Pull request merged**